sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AlterTableAddPartitionExec.scala

```diff
@@ -44,22 +45,25 @@ case class AlterTableAddPartitionExec(
         table.name(), existsParts.map(_.ident), table.partitionSchema())
     }
 
-    notExistsParts match {
-      case Seq() => // Nothing will be done
+    val isTableAltered = notExistsParts match {
+      case Seq() => false // Nothing will be done
       case Seq(partitionSpec) =>
         val partProp = partitionSpec.location.map(loc => "location" -> loc).toMap
         table.createPartition(partitionSpec.ident, partProp.asJava)
+        true
```

Comment:
The v2 API looks inconsistent to me. I do believe either both `dropPartition` and `createPartition` should return nothing (and indicate errors via exceptions) or both methods should return `boolean`.

Reply:
Similar code for partition dropping: https://github.com/apache/spark/blob/84c1f436690c76bfd3bd1a664dba303cfc8381da/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AlterTableDropPartitionExec.scala#L47-L59

Reply:
dropPartition follows dropTable I think

Reply:
probably from Hive?